[MET-2468] Custom payloads, minor fixes, README (#41)
meticalabs/metica-unity-sdk: README.md

# MeticaAPI Unity SDK Guide

**Disclaimer: please note that the SDK is undergoing some quick changes so, for the time being, this information might contain inaccuracies.**

> 

Reviewer: do we need this comment?

| Property                   | Description                                                                                                                                                                                       |
|----------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `apiKey`					 | Your API key

> | `appId`					 | The application identifier that can be configured in Metica's dashboard

Reviewer: to be precise I would write this instead:

```suggestion
| `appId`					 | The application identifier accessible from Metica's dashboard.
```

| appVersion | The game/app version, in [SemVer](https://semver.org/) format                                                                                                                                   | 1.2.3           | 
| locale     | Locale expressed as a combination of language (ISO 639) and country (ISO 3166), // [JDK 8 standard reference](https://www.oracle.com/java/technologies/javase/jdk8-jre8-suported-locales.html). | en-US           |

### Remote Configuration

> The `GetConfig` method can be used to obtain the remote configuration.

Reviewer: ```suggestion
The `GetConfig` method can be used to obtain the smart config.
```

```

### Full State Update

Formerly known as `LogUserAttributes`.

`LogFullStateUpdate` sends a complete snapshot of the user's state to the server, replacing any previously stored data. 
This method fully resets the user's state on the server and expects all relevant state information to be included in the request.
Any user attributes that are currently stored in the server with the given userId but are not sent with this update, will be erased.Logs updates to user attributes and custom user events.

```csharp
Dictionary<string, object> userAttributes = new Dictionary<string, object> { { "level", 25 }, { "favoriteItem", "shield" } };
MeticaAPI.LogFullStateUpdate(userAttributes); 
```

### Partial State Update

`LogPartialStateUpdate` sends a partial update of the user's state to the server
modifying or adding only the provided fields while preserving those that are currently stored on the server.
This method cannot erase existing fields (like `LogFullStateUpdate` does); it can only overwrite values or introduce new ones.

### Custom Payloads

All events support a custom payload that can include any information.  

> It's passed as a `Dictionary<string, object>` to the logging methods. To avoid confusion with other fields, it is recommended to pass the parameter by name, i.g. `LogInstall(customPayload: playerExtraParams)` rather than just `LogInstall(playerExtraParams)`.

Reviewer: ...all events apart from `fullStateUpdate`, `partialStateUpdate` and `CustomEvent`
